### ISSUE TITLE
[QUESTION] Training loss stays stuck above 7

### ISSUE THREAD

**Issue by L-z-Chen** [thu-ml/ReMoE#9]
Hi Ziteng, 

I’m trying out the ReMoE code (https://github.com/thu-ml/ReMoE) 
`bash scripts/train_llama_182m_remoe.sh`
and noticed the training loss stays stuck above `7` even after a long run (screenshot attached) and I ran it for many times, the result doesn't change.
Do you have any ideas on what might be causing this?

Thanks!

<img width="1967" height="879" alt="Image" src="https://github.com/user-attachments/assets/6c49d76e-b4fc-46ad-957c-ab50587e3922" />

**Comment by thuwzt** [comment]
This looks like an environment mismatch in Megatron-LM, especially since the loss curve of your MoE model also appears abnormal. Can you share more details about your software environment (PyTorch/Megatron/TransformerEngine versions, CUDA, NCCL) as well as the hardware setup (GPU type) you are using?

**Comment by L-z-Chen** [comment]
Thanks, Ziteng! Here is my setup:
```
Package                  Version
------------------------ -----------------
annotated-types          0.7.0
apex                     0.1
certifi                  2025.8.3
charset-normalizer       3.4.3
click                    8.2.1
einops                   0.8.1
filelock                 3.19.1
flash_attn               2.8.3
fsspec                   2025.9.0
gitdb                    4.0.12
GitPython                3.1.45
gmpy2                    2.2.1
hf-xet                   1.1.10
huggingface-hub          0.35.0
idna                     3.10
importlib_metadata       8.7.0
Jinja2                   3.1.6
joblib                   1.5.2
MarkupSafe               3.0.2
megatron-core            0.13.1
mpmath                   1.3.0
networkx                 3.5
ninja                    1.13.0
nltk                     3.9.1
numpy                    1.26.4
nvdlfw_inspect           0.1.0
nvidia-cublas-cu12       12.4.2.65
nvidia-cuda-cupti-cu12   12.4.99
nvidia-cuda-nvrtc-cu12   12.4.99
nvidia-cuda-runtime-cu12 12.4.99
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.2.0.44
nvidia-cufile-cu12       1.13.1.3
nvidia-curand-cu12       10.3.5.119
nvidia-cusolver-cu12     11.6.0.99
nvidia-cusparse-cu12     12.3.0.142
nvidia-cusparselt-cu12   0.7.1
nvidia-nccl-cu12         2.20.5
nvidia-nvjitlink-cu12    12.4.99
nvidia-nvtx-cu12         12.4.99
optree                   0.17.0
packaging                25.0
pillow                   11.3.0
pip                      25.2
platformdirs             4.4.0
protobuf                 6.32.1
psutil                   7.1.0
pybind11                 3.0.1
pybind11-global          3.0.1
pydantic                 2.11.9
pydantic_core            2.33.2
PyYAML                   6.0.2
regex                    2025.9.1
requests                 2.32.5
sentry-sdk               2.38.0
setuptools               80.9.0
six                      1.17.0
smmap                    5.0.2
sympy                    1.14.0
torch                    2.6.0
torchaudio               2.4.0+cu124
torchvision              0.21.0+cu124
tqdm                     4.67.1
transformer_engine       2.3.0
triton                   3.2.0+git576374f8
typing_extensions        4.15.0
typing-inspection        0.4.1
urllib3                  2.5.0
wandb                    0.21.4
wheel                    0.45.1
zipp                     3.23.0
```

=== System CUDA ===
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Thu_Mar_28_02:18:24_PDT_2024
Cuda compilation tools, release 12.4, V12.4.131
Build cuda_12.4.r12.4/compiler.34097967_0

=== Hardware ===
index, name, memory.total [MiB], driver_version
0, NVIDIA H100 80GB HBM3, 81559 MiB, 550.90.07
1, NVIDIA H100 80GB HBM3, 81559 MiB, 550.90.07
2, NVIDIA H100 80GB HBM3, 81559 MiB, 550.90.07
3, NVIDIA H100 80GB HBM3, 81559 MiB, 550.90.07
4, NVIDIA H100 80GB HBM3, 81559 MiB, 550.90.07
5, NVIDIA H100 80GB HBM3, 81559 MiB, 550.90.07
6, NVIDIA H100 80GB HBM3, 81559 MiB, 550.90.07
7, NVIDIA H100 80GB HBM3, 81559 MiB, 550.90.07

=== Additional Info ===
OS: Debian GNU/Linux 11 (bullseye)
Kernel: 5.10.0-34-cloud-amd64
CPU: Intel(R) Xeon(R) Platinum 8481C CPU @ 2.70GHz
Total GPUs: 8

**Comment by thuwzt** [comment]
Thanks for the detailed information!

The most likely issue seems to be the version of `transformer_engine`. I recommend downgrading it to `1.9.0` using:
```bash
pip install git+https://github.com/NVIDIA/TransformerEngine.git@e79d915
```

This project relies heavily on TransformerEngine, but in my experience it has limited forward compatibility, so downgrading may help.

For reference, my experimental environment is as follows:
```
Package                  Version       Editable project location
------------------------ ------------- ----------------------------------------
absl-py                  2.1.0
accelerate               1.0.0
aiohappyeyeballs         2.4.0
aiohttp                  3.10.5
aiosignal                1.3.1
annotated-types          0.7.0
apex                     0.1
asttokens                2.4.1
attrs                    24.2.0
certifi                  2024.8.30
chardet                  5.2.0
charset-normalizer       3.3.2
click                    8.1.7
colorama                 0.4.6
comm                     0.2.2
contourpy                1.3.0
cycler                   0.12.1
DataProperty             1.0.1
datasets                 3.0.0
debugpy                  1.8.6
decorator                5.1.1
dill                     0.3.8
docker-pycreds           0.4.0
einops                   0.8.0
evaluate                 0.4.3
exceptiongroup           1.2.2
executing                2.1.0
filelock                 3.15.4
flash_attn               2.5.8
fonttools                4.53.1
frozenlist               1.4.1
fsspec                   2024.6.1
gitdb                    4.0.11
GitPython                3.1.43
grouped_gemm             1.1.4
grpcio                   1.66.1
gviz-api                 1.10.0
huggingface-hub          0.25.0
idna                     3.8
importlib_metadata       8.5.0
ipykernel                6.29.5
ipython                  8.27.0
jedi                     0.19.1
Jinja2                   3.1.4
joblib                   1.4.2
jsonlines                4.0.0
jupyter_client           8.6.3
jupyter_core             5.7.2
kiwisolver               1.4.7
lxml                     5.3.0
Markdown                 3.7
MarkupSafe               2.1.5
matplotlib               3.9.2
matplotlib-inline        0.1.7
mbstrdecoder             1.1.3
more-itertools           10.5.0
mpmath                   1.3.0
multidict                6.1.0
multiprocess             0.70.16
nest_asyncio             1.6.0
networkx                 3.3
ninja                    1.11.1.1
nltk                     3.9.1
numexpr                  2.10.1
numpy                    2.1.1
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-cupti-cu12   12.1.105
nvidia-cuda-nvrtc-cu12   12.1.105
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.0.2.54
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu12     12.1.0.106
nvidia-nccl-cu12         2.20.5
nvidia-nvjitlink-cu12    12.6.68
nvidia-nvtx-cu12         12.1.105
packaging                24.1
pandas                   2.2.2
parso                    0.8.4
pathvalidate             3.2.1
peft                     0.13.1
pexpect                  4.9.0
pickleshare              0.7.5
pillow                   10.4.0
pip                      24.2
platformdirs             4.3.1
portalocker              2.10.1
prompt_toolkit           3.0.48
protobuf                 4.25.4
psutil                   6.0.0
ptyprocess               0.7.0
pure_eval                0.2.3
pyarrow                  17.0.0
pybind11                 2.13.5
pydantic                 2.9.0
pydantic_core            2.23.2
Pygments                 2.18.0
pyparsing                3.1.4
pytablewriter            1.2.0
python-dateutil          2.9.0.post0
pytz                     2024.1
PyYAML                   6.0.2
pyzmq                    26.2.0
regex                    2024.7.24
requests                 2.32.3
rouge_score              0.1.2
sacrebleu                2.4.3
safetensors              0.4.5
scikit-learn             1.5.1
scipy                    1.14.1
sentencepiece            0.2.0
sentry-sdk               2.13.0
setproctitle             1.3.3
setuptools               74.1.2
six                      1.16.0
smmap                    5.0.1
sqlitedict               2.1.0
stack-data               0.6.2
sympy                    1.13.2
tabledata                1.3.3
tabulate                 0.9.0
tcolorpy                 0.1.6
tensorboard              2.17.1
tensorboard-data-server  0.7.2
threadpoolctl            3.5.0
tokenizers               0.20.0
torch                    2.4.1
torch-tb-profiler        0.4.3
torchaudio               2.4.1
torchvision              0.19.1
tornado                  6.4.1
tqdm                     4.66.5
tqdm-multiprocess        0.0.11
traitlets                5.14.3
transformer_engine       1.9.0+e79d915
transformers             4.45.1
triton                   3.0.0
typepy                   1.3.2
typing_extensions        4.12.2
tzdata                   2024.1
urllib3                  2.2.2
wandb                    0.17.9
wcwidth                  0.2.13
Werkzeug                 3.0.4
wheel                    0.44.0
word2number              1.1
xxhash                   3.5.0
yarl                     1.11.1
zipp                     3.20.2
zstandard                0.23.0
```

**Comment by yimeng0701** [comment]
Hi @L-z-Chen, have you solved the problem by downgrading the transformer_engine version? I got the same problem with you...

**Comment by L-z-Chen** [comment]
solved
